src/test/java/com/sugarcrm/candybean/automation/webdriver/WebDriverSystemTest.java

```diff
@@ -171,152 +158,127 @@ public void presentAcceptDismissDialogTest() throws Exception {
 		Thread.sleep(1000);
-		assertEquals(expWindow1Title, iface.wd.getTitle());
-//		iface.interact(iface.getWindowsString());
+		assertEquals(altWindowTitle, iface.wd.getTitle());
 
 		// Close window which should auto-focus to previous window; verify title
 		iface.closeWindow();
-		assertEquals(expWindow0Title, iface.wd.getTitle());
-//		iface.interact(iface.getWindowsString());
+		assertEquals(mainWindowTitle, iface.wd.getTitle());
 
-		// Click pop-up window titled "Tryit Editor v1.8"
-		iface.getWebDriverElement(Strategy.PLINK, "HTML document").click();
-
-		// Navigate elsewhere and trigger popup window
-//		iface.interact("window focus before go: " + iface.wd.getWindowHandle());
-		iface.go(expWindow2URL);
-//		iface.interact("window focus after go: " + iface.wd.getWindowHandle());
-		iface.getWebDriverElement(Strategy.PLINK, "this link").click();
-//		iface.interact(iface.getWindowsString());
-
-		// Verify title with (not) switching to current window by index
-		iface.focusWindow(0);
-		assertEquals(expWindow2Title, iface.wd.getTitle());
-//		iface.interact(iface.getWindowsString());
-
-		// Verify URL with switching to window by title
-		iface.focusWindow(expWindow1Title);
-		String actWindowURL = iface.getURL();
-		assertEquals(expWindow1URL, actWindowURL);
-//		iface.interact(iface.getWindowsString());
+		iface.getWebDriverElement(Strategy.PLINK, "Open in new window").click();
+		iface.focusWindow(altWindowTitle);
+		assertEquals(altWindowTitle, iface.wd.getTitle());
+		assertEquals(testPage2, iface.getURL());
 
 		// Verify URL with switching to window by URL
-		iface.focusWindow(expWindow3URL);
-		assertEquals(expWindow3Title, iface.wd.getTitle());
-//		iface.interact(iface.getWindowsString());
+		iface.focusWindow(testPage1);
+		assertEquals(mainWindowTitle, iface.wd.getTitle());
 
 		// Close window and revert to previous window (1 index); verify URL
 		iface.closeWindow();
-		actWindowURL = iface.getURL();
-		assertEquals(expWindow1URL, actWindowURL);
-//		iface.interact(iface.getWindowsString());
+		assertEquals(testPage2, iface.getURL());
 
-		// Close window and revert to previous window (0 index); verify URL
-		iface.closeWindow();
-		actWindowURL = iface.getURL();
-		assertEquals(expWindow2URL, actWindowURL);
-//		iface.interact(iface.getWindowsString());
-
-		// Verify error by switching to erroneous window titles & indices
-		thrown.expect(Exception.class);
-		thrown.expectMessage("The given focus window string matched no title or URL: garbage");
-		iface.focusWindow("garbage");
-		thrown.expectMessage("Given focus window index is out of bounds: -1 current size: 1");
-		iface.focusWindow(-1);
-		thrown.expectMessage("Given focus window index is out of bounds: 1 current size: 1");
-		iface.focusWindow(1);
-//		iface.interact(iface.getWindowsString());
+		// Verify errors by switching to erroneous window titles & indices
+		// We use try catch rather than expected errors so that we can assert multiple error message
+		try {
```

Comment:
We really shouldn't be checking whether exceptions are thrown 3 times in one test method. So we will need another ticket to address this.